### PR TITLE
Simplification of the code by using glm and reducing templates

### PR DESCRIPTION
**Pull request by FelixTUD**
**BREAKING CHANGES TO THE ISAAC INTERFACE**

- Removed templates for IsaacVisualization class changes expected types to be isaac internal types like isaac_float3
- isaac now expects the render volume to be 3D, the 2D implementation wasn't working before anyway
- Guaranteed internal types simplify the code structure
- Removal of unnecessary templates, as they already had many requirements at the type, which often resulted in complicated compiler errors if they were not met

### CODE REVIEW

File: example/example.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ class TestSource1
         isaac_float3 value = ptr[nIndex.x + nIndex.y * VOLUME_X
                                  + nIndex.z * VOLUME_X * VOLUME_Y];
         isaac_float_dim< 3 > result;
-        result.value = value;
+        result = value;
```
Comment (psychocoderHPC) [inline example/example.cpp]: IMO `result` is not required anymore and you can return `value` directly.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by psychocoderHPC** [comment]
Is it possible to use the upstream version of `glm` for issac. If I remember correctly some changes were required to support CUDA.

**Comment by psychocoderHPC** [comment]
Please add `glm` to the requirements in `INSTALL.md`. If a minimal version is required add this information too.

**Comment by FelixTUD** [comment]
Yes for CUDA a recent glm upstream version works without adjustments, only the future hip support needs some changes

File: example/example.cpp (hunk 1 above)
Comment (psychocoderHPC) [reply example/example.cpp]: @FelixTUD PLease check this comment!

**Comment by FelixTUD** [comment]
@psychocoderHPC the requested changes are integrated

File: example/example.cpp [hunk 2]
```diff
@@ -135,17 +133,14 @@ class TestSource2
     {
         isaac_float value = ptr[nIndex.x + nIndex.y * VOLUME_X
                                 + nIndex.z * VOLUME_X * VOLUME_Y];
-        isaac_float_dim< 1 > result;
-        result.value
-            .x = value;
-        return result;
+        return value;
```
Comment (psychocoderHPC) [inline example/example.cpp]: Here I am not sure if you can return `value` only. Value is a scalar. How is the implementation if you cast `isaac_float` to `isaac_float_dim <3>`? Is the scalar only assigned to `.x` or to all `3` members.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: example/example.cpp (hunk 2 above)
Comment (FelixTUD) [reply example/example.cpp]: It is fixed now, as `feature_dim` is set to `1` the type is `isaac_float_dim <1>` and there is no implicit conversion from `isaac_float` to `isaac_float_dim <1>`

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**